src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -345,6 +347,88 @@ static bool is_input_uniform_constant(shared_ptr<Node> op,
     return rc;
 }
 
+//`simplify_gather`, optimizes gather if Gather is gathering the
+// whole input tensor
+static bool simplify_gather(std::shared_ptr<Node> node)
+{
+    auto gather = as_type_ptr<op::Gather>(node);
+
+    if (gather)
+    {
+        // check if we are gathering the whole input
+        auto data = node->get_argument(0);
+        auto indices = node->get_argument(1);
+
+        // if rank of data and gather output dont match, we will skip
+        if (data->get_shape().size() != node->get_shape().size())
+        {
+            return false;
+        }
+
+        auto axis = gather->get_axis();
+
+        if (axis == std::numeric_limits<int64_t>::max())
+        {
+            NGRAPH_DEBUG << "axis value not set";
+            return false;
+        }
+
+        // lamba function to remove the gather op and rewire the
+        // input data of gather to its users
+        auto replace_gather = [&]() {
```

Comment:
can we use replace_node similar to other optimizations in this pass?

Reply:
https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/graph_util.hpp#L508